compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -334,14 +334,16 @@ class ExtractSemanticDB extends Phase:
        *  the same starting position have the same index.
        */
       def localIdx(sym: Symbol)(using Context): Int =
+        def startPos(sym: Symbol) =
+          if sym.span.exists then sym.span.start else -1
         def computeLocalIdx(): Int =
-          symsAtOffset(sym.span.start).find(_.name == sym.name) match
+          symsAtOffset(startPos(sym)).find(_.name == sym.name) match
             case Some(other) => localIdx(other)
             case None =>
               val idx = nextLocalIdx
               nextLocalIdx += 1
               locals(sym) = idx
-              symsAtOffset(sym.span.start) += sym
+              symsAtOffset(startPos(sym)) += sym
```

Comment:
Does it really make sense to store things at offset -1?

Reply:
Im not sure it matters because this is just a discriminator, but perhaps it is actually a bug in macros because the trees in `sandbox/semanticdb/i8530/App_2.scala` have a position but not their symbol

Reply:
Sounds like it, by default symbols are created without a position so it's easy to forget about passing it

Reply:
what I've seen is that at https://github.com/lampepfl/dotty/blob/363dd03087819d5cfd08b6741cbc75618c17dad9/compiler/src/dotty/tools/dotc/typer/Inliner.scala#L140
when compiling `sandbox/semanticdb/i8530/App_2.scala` is that the symbol of the unapply method being transformed has an index coord

Reply:
ok this is simple, whenever that symbol was loaded `ReadPositions` was not on

Reply:
This is resolved by reading positions from tasty when -Ysemanticdb is set, therefore there is no need to check if span exists